gm4_chairs/data/gm4_chairs/functions/lightning/tag.mcfunction

```diff
@@ -1,6 +1,8 @@
-# runs at player near lightning strike
+# Target NoAI zombified_piglin after lightning strike near player
 # @s = player
+# at @s
 # runs from advancement, gm4_chairs:lightning
 
 advancement revoke @s only gm4_chairs:lightning
-schedule function gm4_chairs:lightning/delay 50t
+
+execute as @e[type=zombified_piglin,nbt={NoAI:1b}] unless data entity @s Tags at @s run function gm4_chairs:lightning/revive_chair
```

Comment:
This no longer verifies that the `zombie_piglin` is actually within a stair block, which is a necessary requirement for being a chair. You should probably include a block check just to be sure.

Reply:
It's possible for the stair block to be broken before the chair is returned, leaving a NoAI piglin. I'm not sure yet if the fire from the lightning can break a wooden stair block within that time, but it would definitely be possible to "abuse" it and create NoAI piglins that won't disappear

Reply:
That is true, but converting all `NoAI:1b` pigman into a chair can also create a lot of issues. Let's say you're setting all mobs in your world to `NoAI:1b` for debug reasons; now you'd have a bunch of chairs in your Pigman farm, that would end up killing themselves.
Therefore, I'd suggest revising this.

Reply:
I guess that's a good reason. Added the stairs check!